src/Workspaces/Core/Portable/FindSymbols/SyntaxTree/SyntaxTreeIndex_Persistence.cs

```diff
@@ -77,24 +66,50 @@ private static bool TryReadVersion(ObjectReader reader, string formatVersion, ou
             return null;
         }
 
-        private static async Task<bool> SaveAsync(
-            Document document, string persistenceName, string formatVersion, SyntaxTreeIndex data, CancellationToken cancellationToken)
+        private static bool FormatAndChecksumMatches(
+            ObjectReader reader, string formatVersion, Checksum checksum)
+        {
+            return TryReadFormatAndChecksum(reader, formatVersion, out var persistChecksum) &&
+                   persistChecksum == checksum;
+        }
+
+        public static async Task<Checksum> GetChecksumAsync(
+            Document document, CancellationToken cancellationToken)
         {
-            Contract.Requires(!await document.IsForkedDocumentWithSyntaxChangesAsync(cancellationToken).ConfigureAwait(false));
+            // Since we build the SyntaxTreeIndex from a SyntaxTree, we need our checksum to change
+            // any time the SyntaxTree could have changed.  Right now, that can only happen if the
+            // text of the document changes, or the ParseOptions change.  So we get the checksums
+            // for both of those, and merge them together to make the final checksum.
+
+            var text = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
you can get these from Document.State.GetStateChecksumAsync().Result.Text and Project.State.GetStateChecksumAsync().Result.ParseOption